Subject: Enabling USB autosuspend for uvc devices
Git-commit 3dae8b41dc5651f8eb22cf310e8b116480ba25b7
From: Matthew Garrett <mjg@redhat.com>
Date:   Thu Sep 16 15:00:04 2010 -0300
Patch-mainline: v2.6.37

    V4L/DVB: uvc: Enable USB autosuspend by default on uvcvideo
    
    We've been doing this for a while in Fedora without any complaints.
    
    Signed-off-by: Matthew Garrett <mjg@redhat.com>
    Acked-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
    Signed-off-by: Mauro Carvalho Chehab <mchehab@redhat.com>
Signed-off-by: Oliver Neukum <oneukum@suse.de>

diff --git a/drivers/media/video/uvc/uvc_driver.c b/drivers/media/video/uvc/uvc_driver.c
index 2ac85d8..93d78f6 100644
--- a/drivers/media/video/uvc/uvc_driver.c
+++ b/drivers/media/video/uvc/uvc_driver.c
@@ -1820,6 +1820,7 @@ static int uvc_probe(struct usb_interface *intf,
 	}
 
 	uvc_trace(UVC_TRACE_PROBE, "UVC device initialized.\n");
+	usb_enable_autosuspend(udev);
 	return 0;
 
 error:
